src/pulse3D/excel_writer.py

```diff
@@ -359,58 +379,53 @@ def _write_xlsx(name: str, metadata_df, continuous_waveforms_df, data, is_optica
-
-        for i, d in enumerate(data):
-            dm = d["metrics"]
-            if dm:
-                force_freq_chart = wb.add_chart({"type": "scatter", "subtype": "straight"})
-                freq_vs_time_chart = wb.add_chart({"type": "scatter", "subtype": "straight"})
-
-                log.info(f'Creating frequency vs time chart for well {d["well_name"]}')
-                create_frequency_vs_time_charts(
-                    freq_vs_time_sheet,
-                    freq_vs_time_chart,
-                    i,
-                    d["well_name"],
-                    dm[1][AMPLITUDE_UUID]["n"],  # number of twitches
-                    list(dm[0]),  # time values
-                    num_metrics,
-                )
-
-                log.info(f'Creating force frequency relationship chart for well {d["well_name"]}')
-                create_force_frequency_relationship_charts(
-                    force_freq_sheet,
-                    force_freq_chart,
-                    i,
-                    d["well_name"],
-                    dm[1][AMPLITUDE_UUID]["n"],  # number of twitches
-                    num_metrics,
-                )
+        log.info("Writing force / frequency charts.")
+        freq_vs_time_sheet = wb.add_worksheet(FORCE_FREQUENCY_RELATIONSHIP_SHEET)
+        force_freq_sheet = wb.add_worksheet(TWITCH_FREQUENCIES_CHART_SHEET_NAME)
+
+        for i,d in enumerate(data):
+            dm = d['metrics']
+            force_freq_chart = wb.add_chart({"type": "scatter", "subtype": "straight"})
+            freq_vs_time_chart = wb.add_chart({"type": "scatter", "subtype": "straight"})
+
+            num_twitches: int = int(dm[1][AMPLITUDE_UUID]["n"][0])
+            time_values: NDArray = d['twitch_time_values']
+
+            log.info(f'Creating frequency vs time chart for well {d["well_name"]}')
+            create_frequency_vs_time_charts(
+                freq_vs_time_sheet,
+                freq_vs_time_chart,
+                i,
+                d['well_name'],
+                num_twitches, #number of twitches
+                time_values, # time values
+                num_metrics,
+            )
+
+            log.info(f'Creating force frequency relationship chart for well {d["well_name"]}')
+            create_force_frequency_relationship_charts(
+                force_freq_sheet,
+                force_freq_chart,
+                i,
+                d['well_name'],
+                num_twitches, #number of twitches
+                num_metrics,
+            )
         log.info(f"Writing {name}")
 
 
-def create_waveform_charts(
-    iter_idx,
-    dm,
-    continuous_waveforms_df,
-    wb,
-    continuous_waveforms_sheet,
-    snapshot_sheet,
-    full_sheet,
-    is_optical_recording,
-):
+def create_waveform_charts(iter_idx, dm, continuous_waveforms_df, wb, continuous_waveforms_sheet, snapshot_sheet, full_sheet, is_optical_recording):
```

Comment:
`is_optical_recording` isn't defined anywhere, in `main` it's a parameter to `_write_xlsx` which is removed in this branch